src/main/java/org/assertj/core/api/IterableAssert.java

```diff
@@ -217,7 +217,7 @@ public int size() {
 
   }
 
-  private static <T> Iterable<T> toIterable(Iterator<T> iterator) {
+  static <T> Iterable<T> toIterable(Iterator<T> iterator) {
```

Comment:
since we have proper Iterator assertions, we can get rid of `LazyIterable` and all the places where is was used.